contracts/strategies/BaseStrategy.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.17;
+
+import "../round/IRoundImplementation.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+/**
+ * @notice Defines the abstract contract for voting algorithms on grants
+ * within a round. Any new voting algorithm would be expected to
+ * extend this abstract contract.
+ * Every BaseStrategy contract would be unique to RoundImplementation
+ * and would be deployed before creating a round
+ */
+abstract contract BaseStrategy is Initializable {
+    // --- Data ---
+
+    /// @notice Round address
+    address payable public roundAddress;
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    // --- Modifier ---
+
+    /// @notice modifier to check if sender is round contract.
+    modifier isRoundContract() {
+        require(
+            roundAddress != address(0),
+            "error: voting contract not linked to a round"
+        );
+        require(
+            msg.sender == roundAddress,
+            "error: can be invoked only by round contract"
+        );
+        _;
+    }
+
+    /// @notice modifier to check if sender is round operator.
+    modifier isRoundOperator() {
+        require(
+            IRoundImplementation(roundAddress).isRoundOperator(msg.sender),
+            "not round operator"
+        );
+        _;
+    }
```

Comment:
Is it possible to have this?:
```solidity
    modifier isRoundOperator(address _caller) {
        require(
            IRoundImplementation(roundAddress).isRoundOperator(_caller),
            "not round operator"
        );
        _;
    }
```

Reply:
Do we need this modifier ? 
If we wanted a msg.sender has the role, we can use the hasRole check instead of adding a custom modifier.

I would say we make `isRoundOperator(address _caller)` an external function instead

Reply:
In this case I would say we need it, yes.
Because it's the strategy calling back the round to validate, if the sender is the round operator.

Reply:
Oh that's a good point!